### ISSUE TITLE
Failed assertion at 'package:cbl_ffi/src/slice.dart'

### ISSUE THREAD

**Issue by nblum37** [cofu-app/cbl-dart#342]
Hi @blaugold,

The current release works fine on nearly all our test mobile phones, except a Samsung Galaxy Xcover 4. For some reason, this mobile phone throws the following error during opening a database:
`SerializationError: No serializer registered for error:'package:cbl_ffi/src/slice.dart': Failed assertion: line 26 pos 41: 'buf != nullptr && size >= 0': is not true.`
If I roll back CBL to a version compatible with Flutter 2.8 and downgrade Flutter to 2.8, everything works fine with the formerly known bugs. We also tested older Samsung phones, all of them do not show this issue. I have currently no clue, what the difference is.

**To Reproduce**

1. Grab a Galaxy Xcover 4 and use any CBL version requiring Flutter 2.10
2. Init CBL with
```
  if (Platform.isIOS || Platform.isAndroid) {
    CblFlutterCe.registerWith();
  }
  await CouchbaseLiteFlutter.init();
```
3. Open a database with `db = await Database.openAsync(databaseName, DatabaseConfiguration(directory: appDocsDir.path));`

**Debug Environment to check the dependency on the Flutter version:**
 - Phone: Galaxy Xcover 4
 - OS: Android 9
 - `cbl`: 1.0.0-beta.12
 - `cbl_flutter`: 1.0.0-beta.10
 - `cbl_flutter_ce`:1.0.0-beta.5

not working / working
 - `Flutter`: 2.10.4 / 2.8.2
We also tested the current CBL release with Flutter 2.10, but it fails as well.

Please let me know, what additional debug information can help you.
Thank you :)

BR,
nblum37

**Comment by blaugold** [comment]
Thanks for reporting this. I have just published `1.1.0` for `cbl`, which includes an improvement to error messages to include a stack trace for this type of error. Could you replicate the issue with this new release and post the new error message?

**Comment by nblum37** [comment]
Unfortunately, there is no remoteStackTrace available. This is all I get in the debugger.
<img width="1899" alt="Bildschirmfoto 2022-04-01 um 05 56 44" src="https://user-images.githubusercontent.com/82163574/161192540-b21a7dd9-222d-4229-9f8a-3be19dd4fe83.png">
Is there another way to get the stacktrace?

**Comment by blaugold** [comment]
You can [evaluate arbitrary expressions](https://www.jetbrains.com/help/idea/examining-suspended-program.html#evaluate-arbitrary) when the debugger is at a breakpoint. Evaluating `exception.remoteStackTrace` should give you the stacktrace.

**Comment by blaugold** [comment]
I found an issue with how errors are rethrown, which results in the loss of the original stack trace. I have a fix for it that you can try with:
```yaml
dependency_overrides:
  cbl:
    git:
      url: https://github.com/cbl-dart/cbl-dart
      ref: fix/full-stack-trace
      path: packages/cbl
```

**Comment by nblum37** [comment]
Hmm, for some reason the `exception.remoteStackTrace` still results into null. Unfortuantely, I dont't see a way to dig deeper into the excepetion.
<img width="1426" alt="Bildschirmfoto 2022-04-01 um 11 13 23" src="https://user-images.githubusercontent.com/82163574/161233569-390a1b08-db16-4e35-8572-52b6c8efb704.png">

**Comment by nblum37** [comment]
Btw, if I start up the app and run over all the errors, after a while I run into a segfault with very limited information. If I never initiate the DBs, then the problem does not occur. But then also a lot of other features are not triggered, so no clue if there is a direct connection. Just as a background information.
<img width="861" alt="Bildschirmfoto 2022-04-01 um 11 29 47" src="https://user-images.githubusercontent.com/82163574/161236933-331ca876-0fbe-44e9-88d5-66bacd219984.png">

**Comment by blaugold** [comment]
Thanks for the additional info. I did find a bug that looks like it could be the cause of this issue, but this bug should only occur when running on 32-bit ARM devices, not 64-bit. I looked at the specs of the Samsung Galaxy Xcover 4, and it has an ARM Cortex-A53 which is 64-bit, it seems like it is running in 32-bit mode, though, looking at the crash back trace.

**Comment by blaugold** [comment]
The bug is in the Dart code which calls into CBL C. `dart:ffi` did not always support variable sized types like `size_t` which is why some CBL C functions had to be wrapped in an C function with statically sized types like `uint_64`. In the migration to using variable sized types directly, I forgot to update some parameter types.

**Comment by blaugold** [comment]
I have published `cbl` `1.1.1` with a fix for this issue. Could you let me know if the issue is solved for you? I was able to replicate it on an older 32-bit device and the fix solved it there.

**Comment by nblum37** [comment]
After some more detailed testing, it really seems, like the issue was resolved 🚀  Thank you very much!!
